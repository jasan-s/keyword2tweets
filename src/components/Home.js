import React from 'react'
import PropTypes from 'prop-types'
import {Helmet} from 'react-helmet'
import { Input } from 'semantic-ui-react'

export default function Home(props) {
  const {userEnteredKeyword, handleInputChange} = props
  return (
    <div>
        <Helmet>
          <title> {process.env.REACT_APP_APP_NAME}</title>
          <meta name='og:url' content='https://Keyword2Tweets.firebaseapp.com'/>
          <meta name='og:type' content='website'/>
          <meta name='og:site_name' content='Keyword2Tweets'/>
          <meta name='og:title' content='Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.'/>
          <meta name='og:description' content='The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.'/>

          <meta name='twitter:card' content='summary'/>
          <meta name='twitter:title' content='Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.'/>
          <meta name='twitter:description' content='The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.'/>
      </Helmet>
        <Input className='search-box' value= {userEnteredKeyword} icon='search'
        onChange={handleInputChange} placeholder='Enter Keyword...' />

    </div>
  )
}

Home.propTypes = {
  userEnteredKeyword: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
}

Home.defaultProps = {
  userEnteredKeyword: '',
}
