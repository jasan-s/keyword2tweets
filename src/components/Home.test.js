import React from 'react'
import { shallow, mount } from 'enzyme'
import Home from './Home'

describe('Home', () => {
	const home = shallow(<Home />)
	it('renders properly', () => {
		expect(home).toMatchSnapshot()
	})
	it('renders a searchbox', () => {
		expect(home.find('.search-box').exists()).toBe(true)
	})
})

describe('When user types in the searchbox', () => {
  const props = {userEnteredKeyword: 'nfl' , handleInputChange: jest.fn()}

  const home = mount(<Home {...props} />)

  const searchBox = home.find('.search-box')
  searchBox.value = 'nfl'
  expect(home.props().userEnteredKeyword).toBe(props.userEnteredKeyword)
})
