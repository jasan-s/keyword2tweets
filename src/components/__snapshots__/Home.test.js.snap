// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    userEnteredKeyword=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
>
          <title>
                     
                    Keyword2Tweets
          </title>
          <meta
                    content="https://Keyword2Tweets.firebaseapp.com"
                    name="og:url"
          />
          <meta
                    content="website"
                    name="og:type"
          />
          <meta
                    content="Keyword2Tweets"
                    name="og:site_name"
          />
          <meta
                    content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                    name="og:title"
          />
          <meta
                    content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                    name="og:description"
          />
          <meta
                    content="summary"
                    name="twitter:card"
          />
          <meta
                    content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                    name="twitter:title"
          />
          <meta
                    content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                    name="twitter:description"
          />
</HelmetWrapper>,
        <Input
          className="search-box"
          icon="search"
          onChange={undefined}
          placeholder="Search..."
          type="text"
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
               
              Keyword2Tweets
</title>,
            <meta
              content="https://Keyword2Tweets.firebaseapp.com"
              name="og:url"
/>,
            <meta
              content="website"
              name="og:type"
/>,
            <meta
              content="Keyword2Tweets"
              name="og:site_name"
/>,
            <meta
              content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
              name="og:title"
/>,
            <meta
              content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
              name="og:description"
/>,
            <meta
              content="summary"
              name="twitter:card"
/>,
            <meta
              content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
              name="twitter:title"
/>,
            <meta
              content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
              name="twitter:description"
/>,
          ],
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Keyword2Tweets",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Keyword2Tweets",
            ],
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "https://Keyword2Tweets.firebaseapp.com",
              "name": "og:url",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "website",
              "name": "og:type",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "Keyword2Tweets",
              "name": "og:site_name",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.",
              "name": "og:title",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.",
              "name": "og:description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "summary",
              "name": "twitter:card",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.",
              "name": "twitter:title",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.",
              "name": "twitter:description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "search-box",
          "icon": "search",
          "onChange": undefined,
          "placeholder": "Search...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
>
            <title>
                         
                        Keyword2Tweets
            </title>
            <meta
                        content="https://Keyword2Tweets.firebaseapp.com"
                        name="og:url"
            />
            <meta
                        content="website"
                        name="og:type"
            />
            <meta
                        content="Keyword2Tweets"
                        name="og:site_name"
            />
            <meta
                        content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                        name="og:title"
            />
            <meta
                        content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                        name="og:description"
            />
            <meta
                        content="summary"
                        name="twitter:card"
            />
            <meta
                        content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                        name="twitter:title"
            />
            <meta
                        content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                        name="twitter:description"
            />
</HelmetWrapper>,
          <Input
            className="search-box"
            icon="search"
            onChange={undefined}
            placeholder="Search..."
            type="text"
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <title>
                 
                Keyword2Tweets
</title>,
              <meta
                content="https://Keyword2Tweets.firebaseapp.com"
                name="og:url"
/>,
              <meta
                content="website"
                name="og:type"
/>,
              <meta
                content="Keyword2Tweets"
                name="og:site_name"
/>,
              <meta
                content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                name="og:title"
/>,
              <meta
                content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                name="og:description"
/>,
              <meta
                content="summary"
                name="twitter:card"
/>,
              <meta
                content="Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter."
                name="twitter:title"
/>,
              <meta
                content="The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist."
                name="twitter:description"
/>,
            ],
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Keyword2Tweets",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Keyword2Tweets",
              ],
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "https://Keyword2Tweets.firebaseapp.com",
                "name": "og:url",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "website",
                "name": "og:type",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "Keyword2Tweets",
                "name": "og:site_name",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.",
                "name": "og:title",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.",
                "name": "og:description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "summary",
                "name": "twitter:card",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "Keyword2Tweets – Given a keyword, retrieves 10 most recet tweets from twitter.",
                "name": "twitter:title",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "The App allows user to enter in a keyword retrieve 10 most recent tweets(using Twitter API) if they exist.",
                "name": "twitter:description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "search-box",
            "icon": "search",
            "onChange": undefined,
            "placeholder": "Search...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
